src/app/modules/webhook/webhook.consumer.ts

```diff
@@ -0,0 +1,247 @@
+      message: 'Webhook queue message contained undefined body',
+      meta: {
+        action: 'createWebhookQueueHandler',
+        sqsMessage,
+      },
+    })
+    // Malformed message will be retried until redrive policy is exceeded,
+    // upon which it will be moved to dead-letter queue
+    return Promise.reject()
+  }
+
+  // Parse message
+  const webhookMessageResult = WebhookQueueMessage.deserialise(Body)
+  if (webhookMessageResult.isErr()) {
+    logger.error({
+      message: 'Webhook queue message could not be parsed',
+      meta: {
+        action: 'createWebhookQueueHandler',
+      },
+      error: webhookMessageResult.error,
+    })
+    return Promise.reject()
+  }
+  const webhookMessage = webhookMessageResult.value
+
+  // If not due, requeue
+  if (!webhookMessage.isDue()) {
+    logger.info({
+      message: 'Webhook not due yet, requeueing',
+      meta: {
+        action: 'createWebhookQueueHandler',
+      },
+    })
+    const requeueResult = await producer.sendMessage(webhookMessage)
+    if (requeueResult.isErr()) {
+      logger.error({
+        message: 'Webhook queue message could not be requeued',
+        meta: {
+          action: 'createWebhookQueueHandler',
+          webhookMessage: webhookMessage.prettify(),
+        },
+        error: requeueResult.error,
+      })
+      // Reject so requeue can be re-attempted
+      return Promise.reject()
+    }
+    // Delete existing message from queue
+    return Promise.resolve()
+  }
+
+  // If due, send webhook
+  // First, retrieve webhook view and URL from database
+  const retryResult = await retrieveWebhookInfo(
+    webhookMessage.submissionId,
+  ).andThen<
+    true,
+    | WebhookRetriesNotEnabledError
+    | WebhookValidationError
+    | WebhookNoMoreRetriesError
+    | WebhookPushToQueueError
+  >((webhookInfo) => {
+    const { webhookUrl, isRetryEnabled } = webhookInfo
+    // Webhook URL was deleted or retries disabled
+    if (!webhookUrl || !isRetryEnabled)
+      return errAsync(
+        new WebhookRetriesNotEnabledError(webhookUrl, isRetryEnabled),
+      )
+
+    // Attempt webhook
+    return WebhookService.sendWebhook(
+      webhookInfo.webhookView,
+      webhookUrl,
+    ).andThen((webhookResponse) => {
+      // Save webhook response to database, but carry on even if it fails
+      void WebhookService.saveWebhookRecord(
+        webhookMessage.submissionId,
+        webhookResponse,
+      )
+
+      // Webhook was successful, no further action required
+      if (isSuccessfulResponse(webhookResponse)) return okAsync(true)
+
+      // Requeue webhook for subsequent retry
+      return webhookMessage
+        .incrementAttempts()
+        .asyncAndThen((newMessage) => producer.sendMessage(newMessage))
+    })
+  })
+
+  if (retryResult.isOk()) return Promise.resolve()
+  // Error cases
+  // Special handling for max retries exceeded - log a separate message
+  // and resolve Promise so that message is removed from queue
```

Comment:
Should there also be a handler for `WebhookRetriesNotEnabledError`?

Reply:
idts, since retries can be re-enabled halfway. wdyt?

Reply:
I see; yea ok. Maybe add a documentation as to why the other errors are not handled then?

Reply:
looking at this again and I think you're right that we should add special handling here. if the admin purposefully disabled retries halfway then there's no reason to move the message to the DLQ. thanks for the catch!